### PR TITLE
Add Travis CI

### PR DESCRIPTION
**Pull request by zingorn**

### CODE REVIEW

**Comment by alexey-sidnev** [comment]
@zingorn , please, use `WIP` prefix in the PR name if you are not ready for review.

**Comment by alexey-trushkov** [comment]
@zingorn seems like travis fails because there is no cuda device on it. To run test on CPU environment variable should be updated `export CUDA_VISIBLE_DEVICES="" `

**Comment by alexey-trushkov** [comment]
@zingorn I believe the problem is that `tensorflow-gpu` required CUDA library to be installed no matter we run code on CPU or GPU. The workaround is to add additional `requirements_ci.txt` file for testing on Travis with only `tensorflow` package as you did earlier: [ae21e4e](https://github.com/opencv/training_toolbox_tensorflow/pull/8/commits/ae21e4ef09620a3b40b8ba27fc9c0f4e3a8adffa).

**Comment by alexey-trushkov** [comment]
Actually `CUDA_VISIBLE_DEVICES` works, but as I mentioned the problem was in `tensorflow-gpu` and `CUDA`.

To use `tensorflow-gpu` `CUDA` must be installed: see [https://www.tensorflow.org/install/gpu](https://www.tensorflow.org/install/gpu). But there is no sense to do that if we don't have GPU device on the machine which is true for Travis. 

So in this case for testing we have to install `tensorflow` cpu version on Travis. To not store two instances of `requirements.txt` probably we can use something like conditional requirements: [https://stackoverflow.com/questions/29222269/is-there-a-way-to-have-a-conditional-requirements-txt-file-for-my-python-applica](https://stackoverflow.com/questions/29222269/is-there-a-way-to-have-a-conditional-requirements-txt-file-for-my-python-applica)

**Comment by zingorn** [comment]
So, done. 
I cannot use requirements.txt conditions because I have no direct access to all environment variables, so lets use setup.py

File: training_toolbox/setup.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+with open('./requirements.txt') as f:
+  required = f.read().splitlines()
+
+# Under the Travis we have to use a CPU version
+if os.environ.get('TRAVIS', None) == 'true':
+  required = [p.replace('-gpu', '') if p.startswith('tensorflow') else p for p in required]
+
+setup(
+    name='Training toolbox',
+    version='1.0.0.dev0',
```
Comment (alexey-sidnev) [inline training_toolbox/setup.py]: Current version is 0.1.0: https://github.com/opencv/training_toolbox_tensorflow/blob/develop/CHANGELOG.md

**Pull request merged**